classes/sdimg.bbclass

```diff
@@ -82,7 +82,10 @@ IMAGE_CMD_sdimg() {
     fi
 
     # Copy boot files to boot partition
-    mcopy -i "${WORKDIR}/fat.dat" -s ${DEPLOY_DIR_IMAGE}/${IMAGE_BOOT_FILES} ::
```

Comment:
Since I'm the author of IMAGE_BOOT_FILES I feel entitled to comment here :) IMO, instead of doing all of this manually  you should use sdimage-bootpart and just add 2 partitions with source set to rootfs. IMAGE_BOOT_FILES will be automagically handled behind the scenes.